Makefile

```diff
@@ -173,7 +173,7 @@ zapai-version: ## prints the zapai version
 
 # Build the delegated IPAM plugin binary.
 azure-ipam-binary:
-	cd $(AZURE_IPAM_DIR) && CGO_ENABLED=0 go build -v -o $(AZURE_IPAM_BUILD_DIR)/azure-ipam$(EXE_EXT) -ldflags "-X main.version=$(AZURE_IPAM_VERSION)" -gcflags="-dwarflocationlists=true"
+	cd $(AZURE_IPAM_DIR) && CGO_ENABLED=0 go build -v -o $(AZURE_IPAM_BUILD_DIR)/azure-ipam$(EXE_EXT) -ldflags "-X buildversion.BuildVersion=$(AZURE_IPAM_VERSION)" -gcflags="-dwarflocationlists=true"
```

Comment:
could just change this to `buildinfo.Version` without renaming/moving the files

Reply:
I was trying to match and see if [this](https://github.com/containernetworking/plugins/blob/v1.2.0/pkg/utils/buildversion/buildversion.go) work but no luck. I did tried populating `buildversion.BuildVersion` directly before but no luck either. The only way I managed to get the version printed out is to have the `version` variable inside `main` then populate it through `main.Version` like CNI and CNS does right now. Is this the only way or I'm I missing something?

Reply:
what you're missing is that `main.version` works in _our_ linker args because `main` is the top level package and we have the `version` var defined there. to inject a value more deeply, we need to fully-qualify the path. 

read the comments in the file you linked and check the mentioned "linker script", you'll see how we need to pass the [full path to the `buildversion` package and variable](https://github.com/containernetworking/plugins/blob/v1.2.0/scripts/release.sh#L10C46-L10C129) for the injection to work. we should not need to change anything except the "-X [...]=$(AZURE_IPAM_VERSION)". our Makefile will automatically set that runtime env to either the bare tag if the current revision is tagged, or the latest tag + dirty hash if the current rev is not tagged.

Reply:
yup, working now